incompatibility/incompatibility-report/chain_test.go

```diff
@@ -0,0 +1,124 @@
+}
+
+// searchForIncompatibility looks for the i1 incompatibilty in the rp2 incompatibility report
+func searchForIncompatibility(i1 *Incompatibility, rp2 *IncompatibilityReport) (found bool) {
+	for _, rp2Item := range rp2.GetIncompatibilities() {
+		if incompatibilityEquality(i1, rp2Item) {
+			found = true
+			return
+		}
+	}
+	return
+}
+
+// incompatibilityReportEquality checks equality for two incompatibility reports
+func incompatibilityReportEquality(rp1 *IncompatibilityReport, rp2 *IncompatibilityReport) (equality bool) {
+	equality = true
+	if len(rp1.GetIncompatibilities()) != len(rp2.GetIncompatibilities()) {
+		equality = false
+		return
+	}
+	for _, incompatibility := range rp1.GetIncompatibilities() {
+		if !searchForIncompatibility(incompatibility, rp2) {
+			equality = false
+			return
+		}
+	}
+	return
+}
+
+func makeIncompatibilityReport(incompatiblities ...*Incompatibility) *IncompatibilityReport {
+	return &IncompatibilityReport{Incompatibilities: incompatiblities}
+}
+
+type InvalidOperationType int
+
+const (
+	OPTIONS = iota
+	HEAD
+	TRACE
+)
+
+func makePathsObject(pathsName string, operationType ...InvalidOperationType) *openapiv3.Paths {
+	var pathItem *openapiv3.PathItem = &openapiv3.PathItem{}
+	for _, opType := range operationType {
+		switch opType {
+		case OPTIONS:
+			pathItem.Options = &openapiv3.Operation{}
+		case HEAD:
+			pathItem.Head = &openapiv3.Operation{}
+		case TRACE:
+			pathItem.Trace = &openapiv3.Operation{}
+		}
+	}
+	return &openapiv3.Paths{Path: []*openapiv3.NamedPathItem{{Name: pathsName, Value: pathItem}}}
+}
+
+// Simple test for in-progress incompatibility chain coverage
+func TestChainCoverage(t *testing.T) {
+
+	var chainTest = []struct {
+		givenDocument                 *openapiv3.Document
+		expectedIncompatibilityReport *IncompatibilityReport
+	}{
+		{&openapiv3.Document{}, &IncompatibilityReport{}},
+		{&openapiv3.Document{Security: []*openapiv3.SecurityRequirement{{
+			AdditionalProperties: []*openapiv3.NamedStringArray{},
+		}}},
+			makeIncompatibilityReport(&Incompatibility{TokenPath: []string{"security"}, Classification: "SECURITY"})},
+		{&openapiv3.Document{Security: []*openapiv3.SecurityRequirement{{
+			AdditionalProperties: []*openapiv3.NamedStringArray{},
+		}}, Paths: makePathsObject("pathName", OPTIONS, HEAD, TRACE)},
+			makeIncompatibilityReport(
+				&Incompatibility{TokenPath: []string{"security"}, Classification: "SECURITY"},
+				&Incompatibility{TokenPath: []string{"paths", "pathName", "options"}, Classification: "OPTIONS"},
+				&Incompatibility{TokenPath: []string{"paths", "pathName", "head"}, Classification: "HEAD"},
+				&Incompatibility{TokenPath: []string{"paths", "pathName", "trace"}, Classification: "TRACE"},
+			)},
+	}
+	for ind, tt := range chainTest {
+		if !incompatibilityReportEquality(SearchChains(tt.givenDocument, IncompatibilityChains...), tt.expectedIncompatibilityReport) {
+			t.Errorf("Unexpected incompatibilty report at index %d", ind)
```

Comment:
Prefer the error form of 

t.Errorf("SearchChains(%v) = %v, want %v", input, got, want) 

go/go-style/decisions#got-before-want

If your comparison/diff facility supports diffs you can print the diff instead of the want and got.  

go/go-style/decisions#print-diffs

Reply:
using diffs